amundsen_application/static/js/components/common/Flag/index.tsx

```diff
@@ -27,6 +28,13 @@ export function convertText(str = '', caseType: string): string {
       return `${str.charAt(0).toUpperCase()}${str.slice(1).toLowerCase()}`;
     case CaseType.UPPER_CASE:
       return str.toUpperCase();
+    case CaseType.TITLE_CASE:
+      const splitStr = str.toLowerCase().split(' ');
+      for (let i = 0; i < splitStr.length; i++) {
+        splitStr[i] =
+          splitStr[i].charAt(0).toUpperCase() + splitStr[i].substring(1);
+      }
+      return splitStr.join(' ');
```

Comment:
I think this should be extracted into an independent, well tested utils function